resources/metadata/mavenStaticCodeChecks.yaml

```diff
@@ -0,0 +1,60 @@
+metadata:
+  name: mavenStaticCodeChecks
+  description: Execute static code checks using maven as is spobugs and pmd
+  longDescription: |
+    Executes spotbugs maven plugin as well as pmd maven plugin for static code checks.
+    SpotBugs is a program to find bugs in Java programs. It looks for instances of “bug patterns” — code instances that are likely to be errors.
+    For more information please visit https://spotbugs.readthedocs.io/en/latest/maven.html
+    PMD is a source code analyzer. It finds common programming flaws like unused variables, empty catch blocks, unnecessary object creation, and so forth. It supports Java, JavaScript, Salesforce.com Apex and Visualforce, PLSQL, Apache Velocity, XML, XSL.
+    For more information please visit https://pmd.github.io/
+spec:
+  inputs:
+    params:
+      - name: mavenModulesExcludes
+        description: Maven modules which should be excluded by the static code checks. By default the modules 'unit-tests' and 'integration-tests' will be excluded.
+        type: '[]string'
+        mandatory: false
```

Comment:
If I am not mistaken, we can now leave out `mandatory: false` statements, as they are the default. (4 more locations)

Reply:
done